src/controllers/auth.controller.js

```diff
@@ -0,0 +1,268 @@
+const bcrypt = require('bcrypt');
+const uuid = require('uuid');
+
+require('dotenv').config();
+
+const emailService = require('../services/email.service');
+const userService = require('../services/user.service');
+const jwtService = require('../services/jwt.service');
+const tokenService = require('../services/token.service');
+const { ApiError } = require('../utils/api.error');
+const { User } = require('../models/user.model');
+
+const register = async (req, res) => {
+  const { name, email, password } = req.body;
+  const activationToken = uuid.v4();
+
+  const { refreshToken } = req.cookies;
+  const userData = await jwtService.validateRefreshToken(refreshToken);
+
+  if (userData) {
+    throw ApiError.badRequest('Only for not authorized');
```

Comment:
The error message 'Only for not authorized' could be improved for clarity. Consider rephrasing it to something like 'Action restricted to unauthorized users only.'